src/main/java/uk/ac/ebi/ega/egacryptor/service/TaskExecutorService.java

```diff
@@ -54,10 +65,15 @@ public void execute(final List<FileToProcess> fileToProcessList) {
      * calculated by application based on no. of cores/processors.
      *
      * @param fileToProcessList List of files to process.
-     * @param noOfThreads No of threads to process list of files.
+     * @param noOfThreads       No of threads to process list of files.
      */
     @Override
     public void execute(final List<FileToProcess> fileToProcessList, final int noOfThreads) {
+        if (fileToProcessList.isEmpty()) {
+            LOGGER.warn(NO_FILES_TO_PROCESS);
+            return;
```

Comment:
We have to return early, otherwise if the fileToProcessList is empty, then noOfThreadsToCreate will be 0, which will make `Executors.newFixedThreadPool(noOfThreadsToCreate)` explode with a totally uninformative `IllegalArgumentException: null` error message. See https://www.ebi.ac.uk/panda/jira/browse/EE-2663